state/store.go

```diff
@@ -23,18 +23,18 @@ import (
 // Store is an interface to perform operations on store.
 type Store interface {
 	BulkStore
-	Init(metadata Metadata) error
-	Features() []Feature
+	Init(ctx context.Context, metadata Metadata) error
+	Features(ctx context.Context) []Feature
```

Comment:
I don't think the Features method should have a context, because it should return a slice synchronously

Reply:
It needs to accept a context because some of the implementations make network connections.

Reply:
Which implementations make a network connection in `Features()`? I would say those implementations are buggy and should be changed. Because of how `Features()` is used, IMHO it should be synchronous.

Reply:
You're quite right. I'm not entirely sure where I got that idea from. Think I am going a bit context mad :smile: